### ISSUE TITLE
Misleading README paragraph about kotlin version

### ISSUE THREAD

**Issue by Mr3zee** [Kotlin/kotlinx-atomicfu#375]
Hi!
I had a problem with kotlin gradle plugin version in my project.
The problem was that I wanted to have version `1.7.22`, and I used it everywhere where I could, but still on evaluation when checking applied plugin version - I had `1.9.0`. Carefully removing all dependencies from my project one by one, I found that `atomicfu:0.22.0` brings this version. 
It is not the problem by itself, but rather the lack of information about such an effect and this confusing paragraph in the README:
```
Starting from version 0.23.1 of the library your project is required to use:
- Gradle 7.0 or newer
- Kotlin 1.7.0 or newer
```
It says that from version `0.23.1` kotlin `1.7.0+` is required and while it is true, the actual kotlin compiler version in the project would not be less than the one `atomicfu` uses.
It would be great to have clarification on possible issue in the README of the project or some other solution
Thank you!

**Comment by Mr3zee** [comment]
Actually, It may be possible to not bring kotlin version at all, for example `kotlinx-serialization` compiler plugin does not bring kotlin compiler with it. But I'm not sure if that would be possible

**Comment by mvicsokolova** [comment]
This issue  will be resolved in the next kotlinx-atomicfu release `0.24.0` (with these changes #406).

Starting from `0.24.0`, `kotlinx-atomicfu` gradle plugin will not add a transitive dependency `org.jetbrains.kotlin:atomicfu:{KGP_version}` to the project, but it will apply the compiler plugin directly with the version of Kotlin Gradle Plugin present in the user project.

Also, please note, that starting from `0.24.0`, `kotlinx-atomicfu` gradle plugin will only be compatible with Kotlin  `1.9.0` or newer. That will be clarified in the README as well.
